### ISSUE TITLE
DW_OP_WASM_location is unsupported

### ISSUE THREAD

**Issue by RReverser** [gimli-rs/gimli#544]
When trying to `dwarfdump` any Wasm file produced with latest Clang or Rust, `DW_AT_frame_base` looks like this:

```
DW_AT_frame_base            len 0x0007: ed03000000009f: WARNING: unsupported operation 0xed
```

Looks like Gimli needs to be taught `DW_OP_WASM_location` as described at https://yurydelendik.github.io/webassembly-dwarf/#DWARF-expressions-and-location-descriptions. /cc @yurydelendik

**Comment by yurydelendik** [comment]
Yes, it will be nice to have Wasm DWARF support. At least under the feature. We are using [crude hack](https://github.com/bytecodealliance/wasmtime/blob/2b325a1878ad5da36c06469046ecf61fddde6c0c/crates/debug/src/transform/expression.rs#L499-L515) in the wasmtime project.

**Comment by philipc** [comment]
Are the semantics of this operation similar to `DW_OP_regx` (rather than `DW_OP_bregx`)?

**Comment by RReverser** [comment]
@philipc I guess depends what you mean by semantics... The attached link likely explains it better than I could.

**Comment by philipc** [comment]
Does it complete the expression or piece (as in `DW_OP_regx`), or does it push a value on the stack (as in `DW_OP_bregx`)? I assume it is the former. Would Wasm ever have a need for the later?

**Comment by RReverser** [comment]
AFAIK it's the former, yes.

**Comment by philipc** [comment]
So having implemented this in #546, I see that llvm is emitting a `DW_OP_stack_value` after it, which I think would only make sense if it pushed a value on the stack. @yurydelendik Can you take a look at this please?

**Comment by RReverser** [comment]
> So having implemented this in #546, I see that llvm is emitting a `DW_OP_stack_value` after it, which I think would only make sense if it pushed a value on the stack. @yurydelendik Can you take a look at this please?

Note that this is being fixed in https://reviews.llvm.org/D94347/new/, which also adds support for indirect values (0x02).

So `DW_OP_WASM_location` will be completing the expression.
